pandas/core/indexes/category.py

```diff
@@ -767,7 +767,8 @@ def _wrap_joined_index(
         self, joined: np.ndarray, other: "CategoricalIndex"
     ) -> "CategoricalIndex":
         name = get_op_result_name(self, other)
-        return self._create_from_codes(joined, name=name)
```

Comment:
you might soon be able to remove _create_from_codes

Reply:
im optimistic